filecache/file_cache_path.py

```diff
@@ -1122,146 +1190,562 @@
+                'is_char_device on a remote file is not implemented')
+
+        return self.as_pathlib().is_char_device()
+
+    def is_fifo(self) -> bool:
+        """Whether this path is a FIFO."""
+
+        if not self.is_local():
+            raise NotImplementedError('is_fifo on a remote file is not implemented')
+
+        return self.as_pathlib().is_fifo()
+
+    def is_socket(self) -> bool:
+        """Whether this path is a socket."""
+
+        if not self.is_local():
+            raise NotImplementedError('is_socket on a remote file is not implemented')
+
+        return self.as_pathlib().is_socket()
+
+    def samefile(self,
+                 other_path: str | Path | FCPath) -> bool:
+        """Return whether other_path is the same or not as this file
+        (as returned by os.path.samefile()).
+        """
+
+        if not isinstance(other_path, FCPath):
+            other_path = FCPath(other_path)
+
+        return self._path == other_path._path
+
+    def absolute(self) -> FCPath:
+        """Return an absolute version of this path
+        No normalization or symlink resolution is performed.
+
+        Use resolve() to resolve symlinks and remove '..' segments.
+        """
+
+        if not self.is_local():
+            return self
+
+        return FCPath(self.as_pathlib().absolute(), copy_from=self)
 
     @classmethod
-    def cwd(cls) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def cwd(cls) -> FCPath:
+        """Return a new path pointing to the current working directory."""
 
-    def expanduser(self) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+        return FCPath(Path.cwd())
+
+    def expanduser(self) -> FCPath:
+        """Return a new path with expanded ~ and ~user constructs
+        (as returned by os.path.expanduser)
+        """
+
+        if self.is_local():
+            return FCPath(self.as_pathlib().expanduser(), copy_from=self)
+
+        return self
 
     @classmethod
-    def home(cls) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def home(cls) -> FCPath:
+        """Return a new path pointing to expanduser('~')."""
 
-    def readlink(self) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+        return FCPath(os.path.expanduser('~'))
+
+    def readlink(self) -> FCPath:
+        """Return the path to which the symbolic link points."""
+
+        if not self.is_local():
+            raise NotImplementedError('readlink on a remote file is not implemented')
+
+        return FCPath(self.as_pathlib().readlink())
 
     def resolve(self,
-                strict: bool = False) -> None:
+                strict: bool = False) -> FCPath:
+        """
+        Make the path absolute, resolving all symlinks on the way and also
+        normalizing it.
+        """
+
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consistent return types for similar methods</b></div>
<div id="fix">

The `resolve()` method's return type has been changed from `None` to `FCPath` but the `realpath()` method still returns `None`. Consider making the return types consistent since both methods serve similar purposes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1376,3 +1376,3 @@
 -    def realpath(self) -> None:
 +    def realpath(self) -> FCPath:
          """Path function not supported by FCPath."""
          raise NotImplementedError
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/32#issuecomment-2558319556>#3d1f03</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged